pyfr/solvers/navstokes/kernels/avis.mako

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+<%inherit file='base'/>
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%
+    import math
+    eps, pi = pyfr.machine_epsilon(fpdtype), math.pi
+%>
+
+<%pyfr:kernel name='avis' ndim='1'
+              s='out view fpdtype_t[${str(nupts)}]'
+              amu_e='out view fpdtype_t[${str(nupts)}]'
+              amu_f='out view fpdtype_t[${str(nfpts)}]'>
+    // Smoothness indicator
+    fpdtype_t totEn = 0.0, pnEn = 0.0, s2;
+
+% for i in range(nupts):
+    s2 = s[${i}]*s[${i}];
+    totEn += s2;
+
+% if ubdegs[i] >= order:
+    pnEn += s2;
+% endif
+% endfor
+
+    // Prevent log(0) for uniform flow
+    fpdtype_t se = (pnEn > 0.0) ? log10(pnEn/totEn) : ${math.log10(eps)};
+
+    fpdtype_t se0= log10(${c['s0']});
+
+    // Compute cell-wise artificial viscosity
+    fpdtype_t mu = (se < se0 - ${c['kappa']}) ? 0.0 : ${0.5*c['amu0']}*(1.0 + sin(${0.5*pi/c['kappa']}*(se - se0)));
+    mu = (se < se0 + ${c['kappa']}) ? mu : ${c['amu0']};
```

Comment:
Might want to consider line-wrapping.  `min/max` may make the code clearer (or it might not!).